### PR TITLE
defaults: change eob (end-of-buffer) from tilde "~" to "middle dot"

### LINKED ISSUE #21694: defaults: change eob (end-of-buffer) from tilde "~" to bullet
### Problem

- The "~" eob fillchar looks very dated.
    - Counterargument: it's distinctive to Vim, familiar, and extremely iconic.
        - Counter-counterargument: Nvim's default UI needs its own subtle distinctiveness.

### Expected behavior

Use bullet char instead of "~".

    set fillchars+=eob:·

- After using it for awhile myself, it doesn't feel garish or overwrought. It feels appropriate: it's a way to give a visual signal without demanding attention.
- The bullet char is already used for Nvim folds, so there's symmetry/precedent.
- It feels heretical, which usually means we should do it.

Compare before/after:

image

### PR DESCRIPTION
**Pull request by ghost**
Resolve #21694.

### CODE REVIEW

**Comment by clason** [comment]
That's some impressive churn for a one-character functional change :D

**Comment by clason** [comment]
Looks like you missed the `oldtest` in `src/nvim/testdir`

**Comment by zeertzjq** [comment]
This is a lot of changes, and this character is also harder to type when writing a test. Maybe change the test setup file instead. For functional tests it's in `test/functional/helpers.lua`. For oldtests it's in `src/nvim/testdir/setup.vim`.

**Review by zeertzjq — CHANGES_REQUESTED** [review]
1. This change is not documented.
2. This character cannot be used if it is double-width. Suggestion below:

File: src/nvim/screen.c [hunk 1]
```diff
@@ -870,7 +870,7 @@ char *set_chars_option(win_T *wp, char **varp, bool apply)
     { &wp->w_p_fcs_chars.foldsep,    "foldsep",   char2cells(0x2502) == 1 ? 0x2502 : '|' },  // │
     { &wp->w_p_fcs_chars.diff,       "diff",      '-' },
     { &wp->w_p_fcs_chars.msgsep,     "msgsep",    ' ' },
-    { &wp->w_p_fcs_chars.eob,        "eob",       '~' },
+    { &wp->w_p_fcs_chars.eob,        "eob",       L'·' },
```
Comment (zeertzjq) [inline src/nvim/screen.c:873]:
```suggestion
    char2cells(0x00b7) == 1 ? 0x00b7 : '~' },  // ·
```

**Review by zeertzjq — CHANGES_REQUESTED** [review]
~~Typed incorrectly again. See below.~~

File: src/nvim/screen.c (hunk 1 above)
Comment (zeertzjq) [inline src/nvim/screen.c:873]:
```suggestion
    { &wp->w_p_fcs_chars.eob,        "eob",       char2cells(0x00b7) == 1 ? 0x00b7 : '~' },  // ·
```

**Review by zeertzjq — CHANGES_REQUESTED** [review]

**Comment by craigmac** [comment]
What's the fallback if the user doesn't have this middle dot character in their chosen font?

**Comment by justinmk** [comment]
> What's the fallback if the user doesn't have this middle dot character in their chosen font?

we don't have a way of checking font, but we can fallback to "~" if char2cells() != 1.

**Comment by benelan** [comment]
I can pick up this PR and work through the review once everyone comes to a consensus in #21694.

I think the Light/Medium/Dark Shade characters are additional good candidates:

```
░ ▒ ▓
```

Here is a screenshot with the `habamax` colorscheme and the Light Shade (U+2591) character, which blends in well:

image


Sorry for commenting on this old PR to circumvent the issue's thread lock :sweat_smile:

**Comment by justinmk** [comment]
> Here is a screenshot with the `habamax` colorscheme and the Light Shade (U+2591) character, which blends in well:

_Interesting_. That actually looks great.

Unlocked https://github.com/neovim/neovim/issues/21694

> once everyone comes to a consensus in https://github.com/neovim/neovim/issues/21694

I'm open to trying the "shade" approach that you showed. Otherwise we should move forward with middle dot, I've been using it without issues.

**Pull request closed without merging**